rustls/src/conn.rs

```diff
@@ -854,9 +849,10 @@ impl<Data> ConnectionCore<Data> {
                 .may_receive_application_data
             && self.common_state.is_tls13()
         {
-            if !is_valid_ccs(&msg)
-                || self.common_state.received_middlebox_ccs > TLS13_MAX_DROPPED_CCS
-            {
+            // We passthrough ChangeCipherSpec messages in the deframer without decrypting them.
+            // Note: this is prior to the record layer, so is unencrypted. See
+            // third paragraph of section 5 in RFC8446.
+            if !msg.is_ccs() || self.common_state.received_middlebox_ccs > TLS13_MAX_DROPPED_CCS {
```

Comment:
Here `msg` is statically of type `BorrowedPlainMessageInbound`, and this is the only call of `is_ccs`. That means, I think, that `is_ccs` can be removed from the `BorrowedPlainMessage` trait.